pkg/controller/nodelifecycle/scheduler/taint_manager.go

```diff
@@ -115,6 +128,27 @@ func deletePodHandler(c clientset.Interface, emitEventFunc func(types.Namespaced
 	}
 }
 
+func addConditionAndDeletePod(ctx context.Context, c clientset.Interface, name, ns string) (err error) {
+	if feature.DefaultFeatureGate.Enabled(features.JobPodFailurePolicy) {
+		pod, err := c.CoreV1().Pods(ns).Get(ctx, name, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		newStatus := pod.Status.DeepCopy()
+		if apipod.UpdatePodCondition(newStatus, &v1.PodCondition{
+			Type:    targetedForDisruption,
+			Status:  v1.ConditionTrue,
+			Reason:  "DeletionByTaintManager",
+			Message: "To be deleted by taint manager due to the NoExecute taint",
+		}) {
+			if _, _, _, err = utilpod.PatchPodStatus(ctx, c, pod.Namespace, pod.Name, pod.UID, pod.Status, *newStatus); err != nil {
```

Comment:
I would prefer to see this code written with SSA (with its client), I think the code would be shorter and have less chance of conflicts, and you can omit the GET. Did you try that?

Reply:
SSA: https://kubernetes.io/docs/reference/using-api/server-side-apply/

Reply:
Haven't tried this yet - I was inspired by the code used to add other pod conditions. Could this be considered as an implementation improvement attempted for Beta? I'm thinking about it due to the time constraint, but would like to try.

Reply:
not sure how relevant this is, but if we were to change it in the next release, the permissions for kcm would have to change a little. In any case, I'm leaving this for @lavalamp to decide if it can be done for beta.

Reply:
Yes, this can wait for beta. I didn't comment on all the other locations, at least some of them could probably also benefit, but at some point it starts being a lot of work.

Reply:
Ok, I've added a it as a graduation criteria for Beta in the KEP update PR: https://github.com/kubernetes/enhancements/pull/3438:
```
Review and implement if feasible adding of pod conditions with the use of [SSA](https://kubernetes.io/docs/reference/using-api/server-side-apply/) client.
```

Reply:
Is there an example of using SSA for get-update in the existing code ?

Reply:
SSA means you can omit the GET.

https://kubernetes.io/blog/2021/08/06/server-side-apply-ga/#server-side-apply-support-in-client-go